File: maxSumSegments.cpp
vector<int> maxSumSegments(vector<int> inputArray) {
    using namespace std;
    vector<int> res;
    int n = inputArray.size();
    for (int len = 1; len <= n; len++) {
        int sum = 0, best, best_i;
        for (int i = 0; i < len; i++) {
            sum += inputArray[i];
        }
        best = sum;
        best_i = 0;
        for (int i = 1; i + len <= n; i++) {
            sum -= inputArray[i - 1];
            sum += inputArray[i + len - 1];
            if (sum > best) {
                best = sum;
                best_i = i;
            }
        }
        res.push_back(best_i);
    }
    return res;
}

File: maxSumSegments.py
def maxSumSegments(inputArray):

    result = []
    for i in range(1, len(inputArray) + 1):
        sum = 0
        mxSum = 0
        index = -1
        for j in range(len(inputArray)):
            sum += inputArray[j]
            if j >= i:
                sum -= inputArray[j - i]
            if j >= i - 1 and (index == -1 or sum > mxSum):
                mxSum = sum
                index = j - i + 1
        result.append(index)
    return result
